Client/BraveSkus/BraveSkusManager.swift

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+import BraveCore
+import BraveVPN
+import os.log
+
+public class BraveSkusManager {
+  private let sku: SkusSkusService
+  
+  public init?(isPrivateMode: Bool) {
+    guard let skusService = Skus.SkusServiceFactory.get(privateMode: isPrivateMode) else {
+      assert(isPrivateMode, "SkusServiceFactory failed to intialize in regular mode, something is wrong.")
+      return nil
+    }
+    
+    self.sku = skusService
+  }
+  
+  public static func refreshSKUCredential(isPrivate: Bool) {
+    guard let _ = Preferences.VPN.skusCredential.value,
+          let domain = Preferences.VPN.skusCredentialDomain.value,
+          let expirationDate = Preferences.VPN.expirationDate.value else {
+      Logger.module.debug("No skus credentials stored in the app.")
+      return
+    }
+    
+    guard expirationDate < Date() else {
+      Logger.module.debug("Existing sku credential has not expired yet, no need to refresh it.")
+      return
+    }
+    
+    guard let manager = BraveSkusManager(isPrivateMode: isPrivate) else {
+      assert(isPrivate, "SkusServiceFactory failed to intialize in regular mode, something is wrong.")
```

Comment:
This assert is redundant, as its already done in the initializer above

Reply:
What i wanted to achieve here is that for `privateMode = true` this manager always returns nil.
I wanted to catch the case where the private mode is false, yet it still returned nil somehow, that's why i throw the assert to catch it early

Reply:
I just mean the assert is the same as the one in the initializer, unless I read it wrong? So the check already exists